src/app/components/feed/FeedInvitation.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { QueryRenderer, graphql } from 'react-relay/compat';
+import Relay from 'react-relay/classic';
+import { browserHistory } from 'react-router';
+import { FormattedMessage } from 'react-intl';
+import cx from 'classnames/bind';
+import ErrorBoundary from '../error/ErrorBoundary';
+import Alert from '../cds/alerts-and-prompts/Alert';
+import styles from './FeedInvitation.module.css';
+import ButtonMain from '../cds/buttons-checkboxes-chips/ButtonMain';
+import ScheduleSendIcon from '../../icons/schedule_send.svg';
+import DoneIcon from '../../icons/done.svg';
+import NotFound from '../NotFound';
+import { can } from '../Can';
+
+const FeedInvitationComponent = ({ routeParams, ...props }) => {
+  // `feed_invitation` is null when attempting to view a feed invitation that doesn't exist or that you don't have permission to view
+  if (!props.feed_invitation) {
+    return <NotFound />;
+  }
```

Comment:
Nice! I'm just curious about why we need to manually check for this instead of having it handled by the API returning a 404 and being automatically redirected by the network layer, like it is in other cases such as attempting to load a team or media without the right permissions.

Reply:
Yeah, for whatever reason we don't get a 404 so this is just me handling it. I could open a bug for it?

Reply:
Sounds like a good idea! I'd rather have this consistently handled by the network layer than having custom approaches in different places.